egui/src/response.rs

```diff
@@ -213,7 +213,9 @@ impl Response {
 
     /// This widget has the keyboard focus (i.e. is receiving key presses).
     pub fn has_focus(&self) -> bool {
-        self.ctx.memory().has_focus(self.id)
+        // Access input and memory in separate statements to prevent deadlock.
+        let has_global_focus = self.ctx.input().raw.has_focus;
+        has_global_focus && self.ctx.memory().has_focus(self.id)
```

Comment:
This makes `Response::has_focus` and `Memory::has_focus` act differently, which seems like a potential footgun.

I'm unsure how to resolve it in a nice way. Perhaps it is good enough to document the difference, i.e. add a docstring to both `Response::has_focus` and `Memory::has_focus` indicating wether or not they are heeding `RawInput::has_focus`.